lib/host.py

```diff
@@ -231,13 +231,15 @@ def execute_script(self, script_contents, shebang='sh', simple_output=True):
             script.write('#!/usr/bin/env ' + shebang + '\n')
             script.write(script_contents)
             script.flush()
-            self.scp(script.name, script.name)
+            remote_path = self.ssh("mktemp").strip()
+            self.scp(script.name, remote_path)
+            self.ssh(['chmod', '0755', remote_path])
```

Comment:
Should probably be moved in try scope in case for some reason a call fails even if unlikely.

Reply:
Sure, have moved them.